controllers/vrg_volrep.go

```diff
@@ -741,6 +741,12 @@ func (v *VRGInstance) reconcileVRForDeletion(pvc *corev1.PersistentVolumeClaim,
 
 			return !requeue
 		}
+
+		if v.deleteSecondaryRestoredPVC(pvc, log) {
+			return requeue
+		}
+
+		return v.undoPVCFinalizersAndPVRetention(pvc, log)
```

Comment:
golangci-lint does not like this:

> Error: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)

But this is bogus error - it is not an error to use if-else, and also the right style for this use case. When we have multiple cases we want to handle using if-else is the right thing. When we want to handle one happy path with one or more exits using if-return is the right thing.

we can change the code to use switch which will be more correct since we really need to check that we are in Primary state in the next block.